src/main/java/tech/stackable/hbase/OpenPolicyAgentAccessController.java

```diff
@@ -0,0 +1,1137 @@
+    LOG.info("preIncrement: user [{}] on table [{}]", user, tableName);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.WRITE);
+    // as per default controller
+    return null;
+  }
+
+  @Override
+  public List<Pair<Cell, Cell>> postIncrementBeforeWAL(
+      ObserverContext<RegionCoprocessorEnvironment> ctx,
+      Mutation mutation,
+      List<Pair<Cell, Cell>> cellPairs) {
+    // we have no ACL table so return as per the similar case in the default controller
+    return cellPairs;
+  }
+
+  @Override
+  public List<Pair<Cell, Cell>> postAppendBeforeWAL(
+      ObserverContext<RegionCoprocessorEnvironment> ctx,
+      Mutation mutation,
+      List<Pair<Cell, Cell>> cellPairs) {
+    // we have no ACL table so return as per the similar case in the default controller
+    return cellPairs;
+  }
+
+  /*********************************** Will be deprecated in 4.0 ***********************************/
+
+  @Override
+  public void grant(
+      RpcController controller,
+      AccessControlProtos.GrantRequest request,
+      RpcCallback<AccessControlProtos.GrantResponse> done) {
+    LOG.debug(
+        "grant for {}/{}", request.getUserPermission().getUser(), request.getUserPermission());
+  }
+
+  @Override
+  public void revoke(
+      RpcController controller,
+      AccessControlProtos.RevokeRequest request,
+      RpcCallback<AccessControlProtos.RevokeResponse> done) {
+    LOG.debug(
+        "revoke for {}/{}", request.getUserPermission().getUser(), request.getUserPermission());
+  }
+
+  @Override
+  public void getUserPermissions(
+      RpcController controller,
+      AccessControlProtos.GetUserPermissionsRequest request,
+      RpcCallback<AccessControlProtos.GetUserPermissionsResponse> done) {}
+
+  @Override
+  public void checkPermissions(
+      RpcController controller,
+      AccessControlProtos.CheckPermissionsRequest request,
+      RpcCallback<AccessControlProtos.CheckPermissionsResponse> done) {}
+
+  @Override
+  public void hasPermission(
+      RpcController controller,
+      AccessControlProtos.HasPermissionRequest request,
+      RpcCallback<AccessControlProtos.HasPermissionResponse> done) {}
+
+  /*********************************** Observer/Service Getters ***********************************/
+
+  @Override
+  public Optional<RegionObserver> getRegionObserver() {
+    return Optional.of(this);
+  }
+
+  @Override
+  public Optional<MasterObserver> getMasterObserver() {
+    return Optional.of(this);
+  }
+
+  @Override
+  public Optional<EndpointObserver> getEndpointObserver() {
+    return Optional.of(this);
+  }
+
+  @Override
+  public Optional<BulkLoadObserver> getBulkLoadObserver() {
+    return Optional.of(this);
+  }
+
+  @Override
+  public Optional<RegionServerObserver> getRegionServerObserver() {
+    return Optional.of(this);
+  }
+
+  /*********************************** Not implemented (yet) ***********************************/
```

Comment:
Do we fall back to the default `AccessController` implementations here?

Edit: I see that some have a LOG statement and others (e.g. preSnapshot) seem to be using the default implementation

Reply:
To be looked at in https://github.com/stackabletech/hbase-opa-authorizer/issues/2